src/components/inclusion-area.ts

```diff
@@ -0,0 +1,104 @@
+import proj4 from 'proj4';
+
+import Map from 'ol/Map.js';
+import { parse } from 'ol/xml';
+import Feature from 'ol/Feature';
+import { Coordinate } from 'ol/coordinate';
+import { Vector } from 'ol/source';
+import { MultiPolygon, Polygon } from 'ol/geom';
+import VectorLayer from 'ol/layer/Vector';
+import { Stroke, Style } from 'ol/style';
+import { GeocityEvent } from '../utils/geocity-event';
+
+export default class InclusionArea {
+  constructor(
+    map: Map,
+    url: string,
+    projectionSource: string,
+    projectionDefinition: string
+  ) {
+    proj4.defs(
+      'EPSG:3857',
+      '+proj=merc +a=6378137 +b=6378137 +lat_ts=0 +lon_0=0 +x_0=0 +y_0=0 +k=1 +units=m +nadgrids=@null +wktext +no_defs +type=crs'
+    );
+    proj4.defs(
+      'SR-ORG:6864',
+      '+proj=merc +lon_0=0 +k=1 +x_0=0 +y_0=0 +a=6378137 +b=6378137 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs'
+    );
+    proj4.defs(projectionSource, projectionDefinition);
+
+    fetch(url)
+      .then((response) => {
+        return response.text();
+      })
+      .then((json) => {
+        const xmlDoc = parse(json);
+        const features = xmlDoc.getElementsByTagName(
+          'wfs:FeatureCollection'
+        )[0];
+        const layers = features.getElementsByTagName('wfs:member');
+        const vectorSource = new Vector();
+        const polygons = [];
+        for (let i = 0; i < layers.length; i++) {
+          const portail = layers[i].getElementsByTagName(
+            'ms:MO_bf_bien_fonds'
+          )[0];
+          if (
+            (portail.getElementsByTagName('ms:genre')[0].innerHTML ===
+              'Domaine public communal' ||
+              portail.getElementsByTagName('ms:genre')[0].innerHTML ===
+                'Domaine public cantonal') &&
+            portail.getElementsByTagName('ms:commune')[0].innerHTML ===
+              'Yverdon-les-Bains'
```

Comment:
too specific to Yverdon

Reply:
Yes, sorry, I was sure that I remove this part yesterday